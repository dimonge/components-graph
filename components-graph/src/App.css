body {
  height: 100vh;
  width: 100vw;
}

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0.5rem 1rem;
  padding: 0.5rem;
}

.header__action {
  display: flex;
  justify-content: space-between;
}

.header__action-search {
  margin-right: 20px;
}
.header__action-searchIcon {
  display: none;
}
@media only screen and (max-width: 320px) {
  .container {
    border: 0;
    margin: 0;
  }
  .header__action-search {
    margin-right: 0;
    display: none;
  }
  h1 {
    font-size: 1rem;
  }
  .header__action-searchIcon {
    display: block;
  }
}
