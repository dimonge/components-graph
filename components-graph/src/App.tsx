import * as React from "react";
import "./App.css";

import data from "./data/components.json";
import Graph from "./components/Graph/index";
import SearchField from "./components/Search/index";
import Profile from "./components/Profile/index";

function App() {
  const [searchText, setSearchText] = React.useState("");

  const onSearchComponents = (value: string): void => {
    setSearchText(value);
  };

  return (
    <div className="container">
      <header className="header">
        <div className="header__title">
          <h1>Capital Flow</h1>
        </div>
        <div className="header__action">
          <SearchField onChange={onSearchComponents} />
          <Profile />
        </div>
      </header>
      <main>
        <Graph data={data} searchText={searchText} />
      </main>
    </div>
  );
}

export default App;
