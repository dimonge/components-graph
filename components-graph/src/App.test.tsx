import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App", () => {
  it("should renders header title", () => {
    render(<App />);
    const title = screen.getByText(/Capital Flow/i);
    expect(title).toBeInTheDocument();
  });

  it("should render the app snapshot", () => {
    const render = renderer.create(<App />);
    expect(render).toMatchSnapshot();
  });

  it("should render the components name", () => {
    render(<App />);
    const componentsName = screen.getByText(/User/i);
    expect(componentsName).toBeInTheDocument();
  });
});
