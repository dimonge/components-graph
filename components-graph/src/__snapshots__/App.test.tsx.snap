// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <div
      className="header__title"
    >
      <h1>
        Capital Flow
      </h1>
    </div>
    <div
      className="header__action"
    >
      <div
        className="header__action-search"
      >
        <div
          className="header__action-searchIcon"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <input
          onChange={[Function]}
          value=""
        />
        <button
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 "
        data-icon="user"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </header>
  <main>
    <div
      className="container"
    />
  </main>
</div>
`;
